app/helpers/application_helper/button/generic_feature_button_with_disable.rb

```diff
@@ -2,19 +2,20 @@ class ApplicationHelper::Button::GenericFeatureButtonWithDisable < ApplicationHe
   needs :@record
 
   def calculate_properties
-    self[:enabled] = !(self[:title] = @error_message if disabled?)
+    super
+    self[:title] = @error_message if disabled?
   end
 
   def disabled?
     begin
       begin
         @error_message = @record.try(:unsupported_reason, @feature)
-      rescue NoMethodError # TODO: remove with deleting AvailabilityMixin module
-        @error_message = @record.try(:is_available_now_error_message, @feature) if @error_message.nil?
+      rescue SupportsFeatureMixin::UnknownFeatureError # TODO: remove with deleting AvailabilityMixin module
+        @error_message = @record.try(:is_available_now_error_message, @feature)
       end
     rescue
```

Comment:
1. you should query the record if the feature is supported and if its not then query for the reason
2. I think the second / nested rescue is too broad.

```ruby
def disabled?
  @error_message = if @record.feature_known?(@feature)
   @record.unsupported_reason(@feature) unless @record.supports?(@feature)
  else
   @record.is_available_now_error_message(@feature)
  end
  @error_message.present?
end
```

I dont know if `@error_message` needs to be assigned during the `disabled?` method, that seems to me like a side effect, but its probably beyond this pr :)